### PR TITLE
Rename instrument type enumeration values

### PR DESCRIPTION
**Pull request by jack-berg**
Rename InstrumentType enum values for consistency with API:
- `OBSERVABLE_SUM` => `OBSERVABLE_COUNTER`
- `OBSERVABLE_UP_DOWN_SUM` => `OBSERVABLE_UP_DOWN_COUNTER` 

InstrumentType is public in the metrics SDK so consistency with the names of the classes in the API matters:
- `ObservableDoubleCounter`
- `ObservableLongCounter`
- `ObservableDoubleUpDownCounter`
- `ObservableLongUpDownCounter`

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
   COUNTER,
   UP_DOWN_COUNTER,
   HISTOGRAM,
-  OBSERVABLE_SUM,
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java:13]: as this is a public API, used for views, I think we should probably go through a deprecation cycle for these values.

Comment (jkwatson) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java:13]: interesting that this doesn't show up in the api diffs. 🤔

Comment (jsuereth) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java:13]: Does the metrics SDK participate in API diffs? I'm not sure if it does.

+1 to deprecation cycle.

**Review by jsuereth — APPROVED** [review]
Agree with John's comment on deprecating.

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java (hunk 1 above)
Comment (jkwatson) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/InstrumentType.java:13]: oh right..it doesn't because it's not stable yet. 🤦🏽

**Pull request merged**